lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -402,66 +470,54 @@ function collectAllStringsInDir(dir) {
 
     const content = fs.readFileSync(absolutePath, 'utf8');
     const exportVars = require(absolutePath);
-    const regexMatches = UISTRINGS_REGEX.test(content);
-    const exportsUIStrings = !!exportVars.UIStrings;
-    if (!regexMatches && !exportsUIStrings) continue;
+    const regexMatch = content.match(UISTRINGS_REGEX);
+    const exportedUIStrings = exportVars.UIStrings;
 
-    if (regexMatches && !exportsUIStrings) {
-      throw new Error('UIStrings defined but not exported');
+    if (!regexMatch) {
```

Comment:
Dunno if it's worth it, but can this be a XOR then a ternary on which error to throw? Ignore if it's pointless, but this kind of boolean logic is deceptively simple, but yet hard to follow.

Reply:
> Dunno if it's worth it, but can this be a XOR then a ternary on which error to throw? Ignore if it's pointless, but this kind of boolean logic is deceptively simple, but yet hard to follow.

that sounds more complicated, but I added a comment and an error message tweak to try to make it clearer :)